File: onyx-database/src/com/onyx/exception/AttributeUpdateException.java
package com.onyx.exception;

/**
 * Created by timothy.osborn on 1/21/15.
 *
 */
public class AttributeUpdateException extends EntityException
{
    public static final String ATTRIBUTE_UPDATE_IDENTIFIER = "Cannot update the entity's identifier";

    protected String attribute;

    /**
     * Constructor with message and attribute
     *
     * @param message Exception message
     * @param attribute Attribute causing exception
     */
    public AttributeUpdateException(String message, String attribute)
    {
        super(message + " : " + attribute);
        this.attribute = attribute;
    }

    @SuppressWarnings("unused")
    public AttributeUpdateException()
    {
        super();
    }
}

File: onyx-database/src/main/kotlin/com/onyx/exception/AttributeUpdateException.kt
package com.onyx.exception

/**
 * Created by timothy.osborn on 1/21/15.
 *
 */
class AttributeUpdateException @JvmOverloads constructor(message: String? = "", attribute: String? = "") : OnyxException(message + " : " + attribute) {

    companion object {
        @JvmField val ATTRIBUTE_UPDATE_IDENTIFIER = "Cannot update the entity's identifier"
    }
}
